planning/mission_planner/src/mission_planner/mission_planner.cpp

```diff
@@ -351,8 +351,47 @@ void MissionPlanner::on_clear_mrm_route(
 
 void MissionPlanner::on_modified_goal(const ModifiedGoal::Message::ConstSharedPtr msg)
 {
-  // TODO(Yutaka Shimizu): reroute if the goal is outside the lane.
-  arrival_checker_.modify_goal(*msg);
```

Comment:
Oh, I think I should not have removed this line. Just add this line with https://github.com/autowarefoundation/autoware.universe/pull/3842/commits/a9ff2aa98f22784f38369eb763443b2becc579b8

Reply:
Note: https://github.com/autowarefoundation/autoware.universe/pull/3842/commits/3f9a6f0f091361b828fec016f40b8d769bace19e
The `modify_goal` is not needed because `set_goai` is called in `change_route`.